File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res, max_sum = nums[0], float('-inf')
        # res = 0

        for i in range(len(nums)):
            max_sum = max(max_sum + nums[i],nums[i])

            res = max(max_sum,res)

        return res
            
            # sum +=nums[i]
            # if sum > max_sum:
                # max_sum = sum
            # if sum < 0:
                # sum = 0
        # return max_sum'''

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int current_sum = 0;
        int max_sum = nums[0];

        for(int n:nums){
            current_sum = max(n,current_sum+n);
            max_sum = max(max_sum,current_sum);
        }
        return max_sum;
    
    }

};